lnrpc/switchrpc/switch_server.go

```diff
@@ -357,6 +359,88 @@ func (s *Server) findEligibleChannelID(pubKey *btcec.PublicKey,
 			amount)
 }
 
+// buildErrorDecryptor constructs an error decrypter given a sphinx session
+// key and hop public keys for a payment route.
+func buildErrorDecryptor(sessionKeyBytes []byte,
```

Comment:
when would we want to give the channelrouter the responsibility to decrypt the error ?

Reply:
Left some more detail here https://github.com/lightningnetwork/lnd/pull/9489#discussion_r2603537791, but I think we may have to allow a channelrouter running remotely to decrypt the error now due to the way the [inputs](https://github.com/lightningnetwork/lnd/blob/v0.20.0-beta/routing/router.go#L126) to the `PaymentAttemptDispatcher` interface's `GetAttemptResult` method are structured.